ebmlite/util.py

```diff
@@ -31,6 +31,56 @@
 # ==============================================================================
 
 
+def createID(idClass, schema, exclude=(), min=0x80, max=0x1FFFFFFF, count=1):
+    """ Generate unique EBML IDs. Primarily intended for use 'offline' by
+        humans creating EBML schemata.
+
+        :param idClass: The EBML class of ID, one of (case-insensitive):
+            * `'a'`: Class A (1 octet, base 0x8X)
+            * `'b'`: Class B (2 octets, base 0x4000)
+            * `'c'`: Class C (3 octets, base 0x200000)
+            * `'d'`: Class D (4 octets, base 0x10000000)
+        :param schema: The `Schema` in which the new IDs must coexist.
+        :param exclude: A list of additional IDs to avoid.
+        :param min: The minimum ID value, within the ID class' range.
+        :param max: The maximum ID value, within the ID class' range.
+        :param count: The maximum number of IDs to generate. The result may be
+            fewer than specified if too few meet the given criteria.
+        :returns: A list of EBML IDs that match the given criteria.
+    """
+    ranges = dict(a = (0x80, 0xFF),
+                   b = (0x4000, 0x7FFF),
+                   c = (0x200000, 0x3FFFFF),
+                   d = (0x10000000, 0x1FFFFFFF))
+    idc = idClass.lower()
+    if idc not in ranges:
+        raise KeyError('Invalid ID class %r: must be one of %r' %
+                       (idClass, list(ranges)))
+
+    # Keep range within the one specified and the one imposed by the ID class
+    idrange = ranges[idc]
+    idrange = (__builtins__['max'](idrange[0], min),
+               __builtins__['min'](idrange[1], max))
+
+    exclude = list(exclude)
+    exclude.extend(schema.elements.keys())
```

Comment:
I'm not sure this *needs* to be changed, but if `exclude` is long then forcing it to be a `list` might make this perform poorly. Can you think of any reason to not instead convert it to a `set`, which has ~O(1) element lookups?

Reply:
Given the nature of the function and the number of elements typically in a schema, I'm not especially concerned about optimization. The function is intended for use by humans, who won't notice (or can live with) a few ms difference.

Reply:
Since I'm going to make a change to `createID()`, so I'm going to go ahead with this suggestion.